src/components/PopupWithForm.js

```diff
@@ -6,6 +6,8 @@ export default class PopupWithForm extends Popup {
     this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._inputList = document.querySelectorAll(".modal__input");
+    this._saveButton = this._popupForm.querySelector(".modal__button-save");
+    this._saveButtonText = this._saveButton.textContent;
   }
 
   _getInputValues() {
```

Comment:
Needs correction (lines 26-27): Сlose modal and reset the form in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.